sklearn/calibration.py

```diff
@@ -526,11 +555,6 @@ def get_metadata_routing(self):
                 splitter=self.cv,
                 method_mapping=MethodMapping().add(callee="split", caller="fit"),
             )
-            # the fit method already accepts everything, therefore we don't
-            # specify parameters. The value passed to ``child`` needs to be the
-            # same as what's passed to ``add`` above, in this case
-            # `"estimator"`.
-            .warn_on(child="estimator", method="fit", params=None)
```

Comment:
we don't need to be backward compatible now when the feature flag is on, so these lines are removed.